tlair/src/main/scala/at/forsyte/apalache/tla/lir/formulas/Base.scala

```diff
@@ -0,0 +1,26 @@
+package at.forsyte.apalache.tla.lir.formulas
+
+import at.forsyte.apalache.tla.lir.formulas.StandardSorts.FunctionSort
+
+abstract class Sort(val sortName: String)
+trait HasSort {
+  val sort: Sort
+}
+
+trait Term extends HasSort
+
+abstract class Variable(name: String) extends Term
+
+object StandardSorts {
+  sealed case class BoolSort() extends Sort("Boolean")
+  sealed case class IntSort() extends Sort("Integer")
+  sealed case class UntypedSort() extends Sort("Untyped")
+  sealed case class UninterpretedSort(override val sortName: String) extends Sort(sortName)
+  sealed case class FunctionSort(to: Sort, from: Sort*) extends Sort("Function") {
+    def arity: Int = from.size
+  }
+}
```

Comment:
>     * What's the value of hiding these case classes inside the `StandardSorts` object? From the usage I see skimming here, I don't see any other sorts defined, so it looks like this creates an additional layer for imports but I do see any clear benefit for that added verbosity. It also doesn't seem necessary for clarity, since every sort has the `sort` suffix in its name).

Fair enough, I'll put them toplevel

>     * Any particular reason why t he two terms `Variable` and `FnTerm` are split by the `StandardSorts`?

Out of habit, from when I was writing c++, I suppose. `FnTerm` depends on `FunctionSort`, so I tend to organize the file in this order.